compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -2977,6 +3199,39 @@ fn eval_bin_op_with_bool_literals(
     Value::Bool(bin_op_result)
 }
 
+fn eval_bin_op_with_double_literals(
+    bin_op: BinOp,
+    lhs_literal: Literal,
+    rhs_literal: Literal,
+    bin_op_expr_span: PackageSpan, // For diagnostic purposes only
+) -> Result<Value, Error> {
+    fn eval_double_div(lhs: f64, rhs: f64, span: PackageSpan) -> Result<Value, Error> {
+        match (lhs, rhs) {
+            (_, 0.0) => Err(EvalError::DivZero(span).into()),
+            (lhs, rhs) => Ok(Value::Double(lhs / rhs)),
+        }
+    }
+
+    // Validate that both literals are doubles.
+    let (Literal::Double(lhs), Literal::Double(rhs)) = (lhs_literal, rhs_literal) else {
+        panic!("at least one literal is not an double: {lhs_literal}, {rhs_literal}");
+    };
+
+    match bin_op {
+        BinOp::Eq => Ok(Value::Bool((lhs - rhs).abs() < f64::EPSILON)),
+        BinOp::Neq => Ok(Value::Bool((lhs - rhs).abs() > f64::EPSILON)),
+        BinOp::Gt => Ok(Value::Bool(lhs > rhs)),
+        BinOp::Gte => Ok(Value::Bool(lhs >= rhs)),
+        BinOp::Lt => Ok(Value::Bool(lhs < rhs)),
+        BinOp::Lte => Ok(Value::Bool(lhs <= rhs)),
```

Comment:
Why does 'eq' and 'neq' allow for an 'epsilon' tolerance in comparisons, but other comparisons such a 'gte' don't? That would seem to lead to some unfortunate asymmetries (e.g. if 'a' is greater than 'b' to within epsilon, then "a == b" could evaluate to true, yet "a <= b" on the same values would evaluate to false).